kernel/src/modelingTools/NonSmoothDynamicalSystem.hpp

```diff
@@ -111,11 +111,7 @@ class NonSmoothDynamicalSystem
   /** add a dynamical system into the DS graph (as a vertex)
    * \param ds a pointer to the system to add
    */
-  inline void insertDynamicalSystem(SP::DynamicalSystem ds)
-  {
-    _topology->insertDynamicalSystem(ds);
-    _mIsLinear = ((ds)->isLinear() && _mIsLinear);
-  };
+  void insertDynamicalSystem(SP::DynamicalSystem ds);
```

Comment:
Is there a change to insertDynamicalSystem other than make it non-inline. I'm by far not against making this one non-inline, espescially when we can get rid of some include in the header files.